master/buildbot/db/pool.py

```diff
@@ -145,8 +146,8 @@ def _start(self):
 
     def _stop(self):
         self._stop_evt = None
+        threads.deferToThreadPool(self.reactor, self._pool, self.engine.dispose)
```

Comment:
The problem is that this will close the engine in the thread pool, but if there is more than one thread, then you will actually close the other engines in the wrong threads